addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -964,6 +977,7 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
     setLabelText: function (previewMode, value, params) {
         this.$target.find('.s_website_form_label_content').text(value);
         if (this._isFieldCustom()) {
+            value = this._getNameWithoutQuotes(value);
```

Comment:
any reason to not insert the "encoded" value in the `s_website_form_label_content` text too so both values are sync'd code wise?
I mean with `.html()` instead of `.text()`

Reply:
@rdeodoo , I probably miss something here but IMO, the problem is that even if we do something like this : 
```diff
     setLabelText: function (previewMode, value, params) {
+        valueEncoded = this._getNameWithoutQuotes(value);
+        this.$target.find('.s_website_form_label_content').html(valueEncoded);
```
With `valueEncoded` set to `L&quot;enfant` for example, the value that will actually be stored in the HTML is `L"enfant` and trying to retrieve this value by doing `const text = this.$target.find('.s_website_form_label_content').html()` will result to `test = L"enfant`. This answers to your last comment:
> I don't understand why we need this and what it does, why is the upfront encoding when the value is set not enough (the other part of the diff)?

Indeed, if the mouse hovers over the "Position" button (see picture), the method `selectLabelPosition` is called. In this method, `field` is  obtained thanks to `this._getActiveField()` and then rendered as an HTML element in `_replaceField` > `_renderField`. The problem is that if `L"enfant` is set as a "Label" for the field form, then `field.name = L"enfant` as `field.name` is set to `this.$target.find('.s_website_form_label_content').text()`. Because `field.name` is not encoded, the `name` attribute of the rendered HTLM element will be set to `L"enfant` but we want the `name` attribute to have the encoded version so this is why the upfront encoding is not enough.

All the problem here is that I did not manage to find a way to retrieve the encoded version of the "Label" from the HTML and in fact, I do not think that is possible as (correct me if I am wrong) it is stored in a decoded version in the HTML even if we do something like `this.$target.find('.s_website_form_label_content').html(valueEncoded)`. But maybe it exists an other way of doing that I don't know :relaxed:  
This also explains why I added 
```diff
+        template.content.querySelectorAll('[name]').forEach(el => el.name = this._getNameWithoutQuotes(el.name));
+        template.content.querySelectorAll('[data-name]').forEach(el => el.dataset.name = this._getNameWithoutQuotes(el.dataset.name));
```
in my commit; So that even if future code uses `.text()` or `.html()` to retrieve the "Label" field name, and re render the new related HTML element (thanks to `_renderField()`), the `name` attribute of the HTML element will correctly contain the encoded version of the "Label" field name.


![image](https://user-images.githubusercontent.com/114914971/233926905-c0187e01-83eb-4ae9-b6e5-3b5eb2692ef0.png)

Reply:
Indeed, testing a bit it doesn't seems possible to write that inside the DOM without it being converted.
I quickly tried by modifying the DOM through the HTML Editor at the time when I asked you to investigate in that regard.

I would then say that the current approach seems to be the best: it's only required/abstracted in the two "low level" form methods, the rest of the code is working as is without any required change.

@qsm-odoo I'd like to have a second opinion at the current approach of the PR. Note that this PR was https://github.com/odoo/odoo/commit/2a7a6151272ba9d23a3362e03d669aef94be9fc4 at first but I [wanted](https://github.com/odoo/odoo/pull/117632#pullrequestreview-1388420277) a low level fix which would not require business code to have to do that: to have the required change abstracted somewhere.

Reply:
@rdeodoo I did not check in details (should I?) but the new solution looks prettier on paper, some thoughts though:
- The old solution I guess "fixed" current databases with the error while the new one will not, right? I guess not very important in this case (We can say "just re-edit the field"?)
- Are those 2 characters really the only problematic ones? In the end why not using encodeURIComponent or something like that?
- I don't see a decode in the PR, does the display of sent information works out-the-box with those encoded characters? Sure we won't see something like "&quot;My Text&quot;" in some mail client / whatever?

Reply:
> The old solution I guess "fixed" current databases with the error while the new one will not, right?

Let's note however that it is currently not possible to have a `"` character as a Label field in a current database (as a traceback directly appears). 

> Are those 2 characters really the only problematic ones? 

It is true that I could also add a rule to replace the "`" character.

> In the end why not using encodeURIComponent or something like that?

Let's note that the problem of `encodeURIComponent()` is that it does not encode single quotes (see https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent#description) and `encodeURIComponent("L'outil")` returns `L'outil`. The use of this function would work with the current code and in the future, this could work as long as future code does not do something like this: ```this.$target[0].querySelector(`[name='${name}']`)```. After some researches, I did not find an existing function that could be used to replace all `"`, `'` and ``` ` ``` characters

> I don't see a decode in the PR, does the display of sent information works out-the-box with those encoded characters? Sure we won't see something like ""My Text"" in some mail client / whatever?

It seems that due to the use of the character entities, the clients will see the `"` and `'` characters in their mails (see picture).

![image](https://user-images.githubusercontent.com/114914971/235873947-3c3b5b1f-8da7-48fa-86ff-3462125a814d.png)

Reply:
> Let's note however that it is currently not possible to have a " character as a Label field in a current database (as a traceback directly appears).

That's not true? :thinking: Or am I am missing something? There is no traceback during edition: you can save, then it crashes on every page load and the customer is stuck forever because he cannot re-edit the page (even after your fix). His only possibility is to delete the page or edit it via HTML editor. I'll let @rdeodoo decide if that's ok to him, I am not sure myself, 50-50: probably not many users have this problem as blocking otherwise we would have many tickets but... don't know :wink: 

> It is true that I could also add a rule to replace the "`" character.

If that's the only one, it's ok but I was thinking any special character could be a problem.

> It seems that due to the use of the character entities, the clients will see the " and ' characters in their mails (see picture).

As long as you are sure, fine by me :+1:

Reply:
>  I'll let @rdeodoo decide

Let's not focus too much on existing customers having the possible issues this time.
(I would have been more happy if we actually had a test coming with the fix rather than something working for those users :( try to always have one, I think here it's not that difficult if I am not wrong we have a tour testing the visibility? would be some extra steps to add)

> I was thinking any special character could be a problem

Could be at some point if we were to change some selector but not now.
![image](https://github.com/odoo/odoo/assets/30048408/371413a8-105c-42e1-9398-2091088d946b)
If we were to remove the `"` usage in the selector, it would break if the user typed special char, see
![image](https://github.com/odoo/odoo/assets/30048408/7e5d115d-f9b5-4a3f-8fa7-a9aa4b484fe2)

At this point, I am still not convinced and I asked your opinion because.. It really looks weird.
It really doesn't feel robust.
We should probably tackle the problem another way: instead of a full jquery selector, can't you @loco-odoo simply loop through all inputs and perform a check on the property against the value?
Eg, instead of the current approach
```js
value = this._getQuotesEncodedName(value); // You patch this
// [..]
const dependentEls = this.formEl.querySelectorAll(`.s_website_form_field[data-visibility-dependency="${previousInputName}"]`); // To make this selector works
```
Do something like
```js
// leave `value` untouched
const dependentEls = this.formEl.querySelectorAll(`.s_website_form_field[data-visibility-dependency]`);
// filter in a for loop the `dependentEls` -> `el => el.dataset.visibilityDependency == value`
```

Reply:
You are right.
Keeping the current implementation there doesn't seem to be a good solution.
This should be refactored to not rely on what the user is typing but on something else, somehow. Like a new DOM attribute for visibility, basically a generated ID "visibility-id-34343".
We should dissociate the label (user typed string without restriction) from this technical behavior/selector mechanism.

This is probably something to do in master one day.
Let's keep what you did and hope for the best. Should it fail someday (due to new code as our current codebase will be fine) we will spend time on this.